Source/Scene/Scene.js

```diff
@@ -2696,13 +2750,166 @@ define([
         return object;
     };
 
+    var fragDepthRegex = /\bgl_FragDepthEXT\b/;
+    var discardRegex = /\bdiscard\b/;
+
+    function getDepthOnlyShaderProgram(scene, context, shaderProgram) {
```

Comment:
Perhaps add a comment here or somewhere that this is an optimization because we're not sure that drivers will replace the fragment shader when it doesn't discard/write-depth when the color mask is all falses and the framebuffer doesn't have a color attachment.

Who knows in the future, this might not be required.